### PR TITLE
refactor(LimitedChaosMonkey): add an attribute to filter the nemesis

### PR DESCRIPTION
**Pull request by amoskong**
Ticket: https://trello.com/c/lFhSinLV/1790-redefine-limited-monkey-using-an-attribute

Currently we have a fixed list of nemesis in LimitedChaosMonkey. This
patch changed it to filter the nemesis by the 'limited' attribute.

Signed-off-by: Amos Kong <amos@scylladb.com>

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by amoskong** [comment]
Tested this PR locally by docker backend.

`sdcm.nemesis.LimitedChaosMonkey: ['disrupt_stop_wait_start_scylla_server', 'disrupt_stop_start_scylla_server', 'disrupt_hard_reboot_node', 'disrupt_soft_reboot_node', 'disrupt_nodetool_drain', 'disrupt_nodetool_decommission', 'disrupt_no_corrupt_repair', 'disrupt_major_compaction', 'disrupt_nodetool_refresh', 'disrupt_nodetool_enospc', 'disrupt_nodetool_cleanup', 'disrupt_truncate', 'disrupt_modify_table', 'disrupt_add_drop_column', 'disrupt_mgmt_backup', 'disrupt_mgmt_backup_specific_keyspaces', 'disrupt_mgmt_repair_cli', 'disrupt_abort_repair', 'disrupt_snapshot_operations']`

Note: 3 nemesises were filtered out because they aren't supported by docker backend.

File: sdcm/nemesis.py [hunk 1]
```diff
@@ -3358,6 +3376,10 @@ def _random_disrupt(self):
 
 class LimitedChaosMonkey(Nemesis):
 
+    def __init__(self, *args, **kwargs):
+        super(LimitedChaosMonkey, self).__init__(*args, **kwargs)
```
Comment (dkropachev) [inline sdcm/nemesis.py:3380]: `super().__init__(...)`

File: sdcm/sct_events/base.py [hunk 2]
```diff
@@ -203,8 +203,11 @@ def __eq__(self, other):
 
     def __del__(self):
         if self._ready_to_publish:
-            LOGGER.warning(
-                "[SCT internal warning] %s has not been published or dumped, maybe you missed .publish()", self)
+            warning = f"[SCT internal warning] {self} has not been published or dumped, maybe you missed .publish()"
+            try:
+                LOGGER.warning(warning)
+            except:
+                print(warning)
```
Comment (dkropachev) [inline sdcm/sct_events/base.py:210]: rebase to master ?

**Review by dkropachev — DISMISSED** [review]

**Comment by dkropachev** [comment]
fix 1 issue rebase to the master and good to go

File: sdcm/sct_events/base.py (hunk 2 above)
Comment (amoskong) [reply sdcm/sct_events/base.py]: Done.

File: sdcm/nemesis.py (hunk 1 above)
Comment (amoskong) [reply sdcm/nemesis.py]: Fixed, thanks!

**Comment by amoskong** [comment]
Updated the PR to address @dkropachev 's comment.

File: sdcm/nemesis.py [hunk 3]
```diff
@@ -297,12 +298,14 @@ def get_list_of_methods_compatible_with_backend(
...
             run_with_gemini: Optional[bool] = None,
-            networking: Optional[bool] = None) -> List[str]:
+            networking: Optional[bool] = None,
+            limited: Optional[bool] = None) -> List[str]:
         return self.get_list_of_methods_by_flags(
             disruptive=disruptive,
             run_with_gemini=run_with_gemini,
             networking=networking,
-            kubernetes=True if self._is_it_on_kubernetes() else None
+            kubernetes=(True if self._is_it_on_kubernetes() else None),
```
Comment (dkropachev) [inline sdcm/nemesis.py:307]: Let's replace it with :
`kubernetes=self._is_it_on_kubernetes() or None,`

**Review by dkropachev — APPROVED** [review]
One more optional thing that could be fixed

File: sdcm/nemesis.py (hunk 3 above)
Comment (amoskong) [reply sdcm/nemesis.py]: Fixed.

File: sdcm/nemesis.py [hunk 4]
```diff
@@ -297,12 +298,14 @@ def get_list_of_methods_compatible_with_backend(
...
             run_with_gemini: Optional[bool] = None,
-            networking: Optional[bool] = None) -> List[str]:
+            networking: Optional[bool] = None,
+            limited: Optional[bool] = None) -> List[str]:
         return self.get_list_of_methods_by_flags(
             disruptive=disruptive,
             run_with_gemini=run_with_gemini,
             networking=networking,
-            kubernetes=True if self._is_it_on_kubernetes() else None
+            kubernetes=(self._is_it_on_kubernetes() or None),
```
Comment (dkropachev) [inline sdcm/nemesis.py:307]: Redundant parentheses

**Review by dkropachev — CHANGES_REQUESTED** [review]

File: sdcm/nemesis.py (hunk 4 above)
Comment (amoskong) [reply sdcm/nemesis.py:307]: The parentheses isn't necessary, it enhances the readability of the code.
I'm fine to remove them.

**Review by dkropachev — APPROVED** [review]

**Review by roydahan — APPROVED** [review]

**Pull request merged**